pkg/resources/digitalocean/cloud.go

```diff
@@ -175,3 +183,97 @@ func (c *Cloud) GetApiIngressStatus(cluster *kops.Cluster) ([]kops.ApiIngressSta
 
 	return nil, nil
 }
+
+// FindClusterStatus discovers the status of the cluster, by looking for the tagged etcd volumes
+func (c *Cloud) FindClusterStatus(cluster *kops.Cluster) (*kops.ClusterStatus, error) {
+	etcdStatus, err := findEtcdStatus(c, cluster)
+	if err != nil {
+		return nil, err
+	}
+	status := &kops.ClusterStatus{
+		EtcdClusters: etcdStatus,
+	}
+	klog.V(2).Infof("Cluster status (from cloud): %v", fi.DebugAsJsonString(status))
+	return status, nil
+}
+
+// findEtcdStatus discovers the status of etcd, by looking for the tagged etcd volumes
+func findEtcdStatus(c *Cloud, cluster *kops.Cluster) ([]kops.EtcdClusterStatus, error) {
+	statusMap := make(map[string]*kops.EtcdClusterStatus)
+	volumes, err := getAllVolumesByRegion(c, c.RegionName)
+
+	if err != nil {
+		return nil, fmt.Errorf("error describing volumes: %v", err)
```

Comment:
Maybe use a error message closer to the function name we called and include the region name, e.g.:

```go
return nil, fmt.Errorf("failed to get all volumes by region from %s: %v", c.RegionName, err)
```

Reply:
sure, updated.